internal/controllers/seller/update_test.go

```diff
@@ -0,0 +1,278 @@
+					wanted: wanted{
+						calls:      1,
+						statusCode: http.StatusConflict,
+						message:    "1062",
+					},
+				},*/
+		{
+			name: "422 - When passing a body with empty fields",
+			id:   "1",
+			sellerJSON: `{
+				"id": 1,
+				"cid": "",
+				"company_name": "",
+				"address": "",
+				"telephone": "",
+				"locality_id": 0
+			}`,
+			callErr: nil,
+			wanted: wanted{
+				statusCode: http.StatusUnprocessableEntity,
+				message:    "error: attribute CID cannot be empty",
+			},
+		},
+		{
+			name: "404 - When the repository don't find the seller",
+			id:   "1",
+			sellerJSON: `{
+				"id": 1,
+				"cid": "123",
+				"company_name": "Meli",
+				"address": "123 Main St",
+				"telephone": "123-456-7890",
+				"locality_id": 1
+			}`,
+			callErr: models.ErrSellerNotFound,
+			wanted: wanted{
+				calls:      1,
+				statusCode: http.StatusNotFound,
+				message:    "seller not found",
+			},
+		},
+		{
+			name: "500 - When the repository returns an error",
+			id:   "1",
+			sellerJSON: `{
+						"id": 1,
+						"cid": "123",
+						"company_name": "Meli",
+						"address": "123 Main St",
+						"telephone": "123-456-7890",
+						"locality_id": 1
+					}`,
+			callErr: models.ErrorNoChangesMade,
+			wanted: wanted{
+				calls:      1,
+				statusCode: http.StatusBadRequest,
+				message:    "no changes made",
+			},
+		},
+		{
+			name: "500 - When the repository returns an error",
+			id:   "1",
+			sellerJSON: `{
+						"id": 1,
+						"cid": "123",
+						"company_name": "Meli",
+						"address": "123 Main St",
+						"telephone": "123-456-7890",
+						"locality_id": 1
+					}`,
+			callErr: &mysql.MySQLError{Number: mysqlerr.CodeDuplicateEntry},
+			wanted: wanted{
+				calls:      1,
+				statusCode: http.StatusConflict,
+				message:    "1062",
+			},
+		},
+		{
+			name: "500 - When the repository returns an error",
+			id:   "1",
+			sellerJSON: `{
+						"id": 1,
+						"cid": "123",
+						"company_name": "Meli",
+						"address": "123 Main St",
+						"telephone": "123-456-7890",
+						"locality_id": 1
+					}`,
+			callErr: errors.New("Generic error"),
+			wanted: wanted{
+				calls:      1,
+				statusCode: http.StatusInternalServerError,
+				message:    "Generic error",
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Arrange
+			rp := sellersrp.NewSellersRepositoryMock()
+			sv := service.NewSellersService(rp)
+			ctl := sellersctl.NewSellersController(sv)
+
+			r := chi.NewRouter()
+			r.Patch("/api/v1/sellers/{id}", ctl.Update)
+			url := "/api/v1/sellers/" + string(tt.id)
+			req := httptest.NewRequest(http.MethodPatch, url, strings.NewReader(tt.sellerJSON))
+			res := httptest.NewRecorder()
+
+			// Act
+			rp.On(
+				"Update",
+				mock.AnythingOfType("int"),
+				mock.AnythingOfType("models.SellerDTO"),
+			).Return(tt.wanted.seller, tt.callErr)
+			r.ServeHTTP(res, req)
+			ctl.Update(res, req)
+
```

Comment:
quando você chama o serve http, você não precisa chamar a ação do controller também
```suggestion
			r.ServeHTTP(res, req)

```